lite/examples/object_detection/ios/ObjectDetection/ModelDataHandler/ModelDataHandler.swift

```diff
@@ -121,122 +109,53 @@ class ModelDataHandler: NSObject {
-    let scaledSize = CGSize(width: inputWidth, height: inputHeight)
-    guard let scaledPixelBuffer = pixelBuffer.resized(to: scaledSize) else {
-      return nil
-    }
 
-    let interval: TimeInterval
-    let outputBoundingBox: Tensor
-    let outputClasses: Tensor
-    let outputScores: Tensor
-    let outputCount: Tensor
+    guard let mlImage = MLImage(pixelBuffer: pixelBuffer) else { return nil }
+    // Run inference
     do {
-      let inputTensor = try interpreter.input(at: 0)
-
-      // Remove the alpha component from the image buffer to get the RGB data.
-      guard let rgbData = rgbDataFromBuffer(
-        scaledPixelBuffer,
-        byteCount: batchSize * inputWidth * inputHeight * inputChannels,
-        isModelQuantized: inputTensor.dataType == .uInt8
-      ) else {
-        print("Failed to convert the image buffer to RGB data.")
-        return nil
-      }
-
-      // Copy the RGB data to the input `Tensor`.
-      try interpreter.copy(rgbData, toInputAt: 0)
-
-      // Run inference by invoking the `Interpreter`.
       let startDate = Date()
-      try interpreter.invoke()
-      interval = Date().timeIntervalSince(startDate) * 1000
+      let detectionResult = try detector.detect(gmlImage: mlImage)
+      let interval = Date().timeIntervalSince(startDate) * 1000
+      let inferences = formatResults(detectionResult: detectionResult)
 
-      outputBoundingBox = try interpreter.output(at: 0)
-      outputClasses = try interpreter.output(at: 1)
-      outputScores = try interpreter.output(at: 2)
-      outputCount = try interpreter.output(at: 3)
+      // Returns the inference time and inferences
+      let result = Result(inferenceTime: interval, inferences: inferences)
+      return result
     } catch let error {
       print("Failed to invoke the interpreter with error: \(error.localizedDescription)")
       return nil
     }
-
-    // Formats the results
-    let resultArray = formatResults(
-      boundingBox: [Float](unsafeData: outputBoundingBox.data) ?? [],
-      outputClasses: [Float](unsafeData: outputClasses.data) ?? [],
-      outputScores: [Float](unsafeData: outputScores.data) ?? [],
-      outputCount: Int(([Float](unsafeData: outputCount.data) ?? [0])[0]),
-      width: CGFloat(imageWidth),
-      height: CGFloat(imageHeight)
-    )
-
-    // Returns the inference time and inferences
-    let result = Result(inferenceTime: interval, inferences: resultArray)
-    return result
   }
 
-  /// Filters out all the results with confidence score < threshold and returns the top N results
+  /// Filters out all the results with confidence score < threshold and returns the results
   /// sorted in descending order.
-  func formatResults(boundingBox: [Float], outputClasses: [Float], outputScores: [Float], outputCount: Int, width: CGFloat, height: CGFloat) -> [Inference]{
+  ///
+  func formatResults(detectionResult: DetectionResult) -> [Inference] {
```

Comment:
Remove the `Inference` class and use DetectionResult directly to plot in the view controller to plot the results.